core/tx_pool.go

```diff
@@ -1503,9 +1504,22 @@ func (pool *TxPool) executableTxFilter(costLimit *big.Int) func(tx *types.Transa
 // equal number for all for accounts with many pending transactions.
 func (pool *TxPool) truncatePending() {
 	pending := uint64(0)
+
+	parent := pool.chain.CurrentBlock()
+	l1BaseFee := fees.GetL1BaseFee(pool.currentState)
```

Comment:
this might introduce overhead of state access

Reply:
fixed [c5df443](https://github.com/scroll-tech/go-ethereum/pull/983/commits/c5df443e4b640ff6d531f5365f6f3420b2b1d793)